### PR TITLE
Updating the MAINTAINERS.md file

### LINKED ISSUE #31: Baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions
Follow https://github.com/opensearch-project/.github/issues/125 to baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions.

Close this issue when:

1. [MAINTAINERS.md](MAINTAINERS.md) has the correct list of project maintainers.
2. [CODEOWNERS](CODEOWNERS) exists and has the correct list of aliases.
3. Repo permissions only contain individual aliases as collaborators with maintain rights, admin, and triage teams. 
4. All other teams are removed from repo permissions.

If this repo's permissions was already baselined, please confirm the above when closing this issue.

### PR DESCRIPTION
**Pull request by bbarani**
### Description
Updated the MAINTAINERS.md file 

### Issues Resolved
Closes #31 

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by bbarani** [comment]
@SuZhou-Joe Please merge this PR when you get time. I have fixed permissions in this repo.

File: MAINTAINERS.md [hunk 1]
```diff
@@ -3,16 +3,17 @@
...
+| Maintainer            | GitHub ID                                                | Affiliation |
+|-----------------------|----------------------------------------------------------| ----------- |
+| Saurabh Singh         | [getsaurabh02](https://github.com/getsaurabh02)          | Amazon |
+| Subhobrata Dey        | [sbcd90](https://github.com/sbcd90)                      | Amazon |
+| Praveen Sameneni      | [praveensameneni](https://github.com/praveensameneni)    | Amazon |
+| Xuesong Luo           | [xluo-aws](https://github.com/xluo-aws)       | Amazon      |
+| Hailong Cui           | [Hailong-am](https://github.com/Hailong-am)   | Amazon      |
+| Binlong Gao           | [gaobinlong](https://github.com/gaobinlong)   | Amazon      |
+| Zhou Su               | [SuZhou-Joe](https://github.com/SuZhou-Joe)   | Amazon      |
+| Yuye Zhu              | [SuZhou-Joe](https://github.com/yuye-aws)   | Amazon      |
```
Comment (xluo-aws) [inline MAINTAINERS.md:15]: Github ID should be yuye-aws, not SuZhou-Joe

File: MAINTAINERS.md [hunk 2]
```diff
@@ -3,16 +3,17 @@
...
+| zhichao               | [SuZhou-Joe](https://github.com/zhichao-aws)   | Amazon      |
```
Comment (xluo-aws) [inline MAINTAINERS.md:16]: Name is Zhichao Geng, Github id should be zhichao-aws, not SuZhou-Joe

File: MAINTAINERS.md (hunk 1 above)
Comment (SuZhou-Joe) [inline MAINTAINERS.md:15]:
```suggestion
| Yuye Zhu              | [yuye-aws](https://github.com/yuye-aws)   | Amazon      |
```

File: MAINTAINERS.md (hunk 2 above)
Comment (SuZhou-Joe) [inline MAINTAINERS.md:16]:
```suggestion
| zhichao               | [zhichao-aws](https://github.com/zhichao-aws)   | Amazon      |
```

**Review by SuZhou-Joe — CHANGES_REQUESTED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Review by xluo-aws — APPROVED** [review]

**Pull request merged**